devicetypes/zooz/zooz-double-switch-zen30.src/zooz-double-switch-zen30.groovy

```diff
@@ -0,0 +1,479 @@
+	])
+}
+
+String multiChannelCmdEncapCmd(cmd, endpoint) {
+	if (endpoint) {
+		return secureCmd(zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint:safeToInt(endpoint)).encapsulate(cmd))
+	} else {
+		return secureCmd(cmd)
+	}
+}
+
+String secureCmd(cmd) {
+	if (zwaveInfo?.zw?.contains("s")) {
+		return zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+	} else {
+		return cmd.format()
+	}
+}
+
+def parse(String description) {
+	def cmd = zwave.parse(description, commandClassVersions)
+	if (cmd) {
+		zwaveEvent(cmd)
+	} else {
+		log.warn "Unable to parse: $description"
+	}
+	return []
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd) {
+	// Workaround that was added to all SmartThings Multichannel DTHs.
+	if ((cmd.commandClass == supervisionCC) && (cmd.parameter.size >= 4)) { // Supervision encapsulated Message
+		// Supervision header is 4 bytes long, two bytes dropped here are the latter two bytes of the supervision header
+		cmd.parameter = cmd.parameter.drop(2)
+		// Updated Command Class/Command now with the remaining bytes
+		cmd.commandClass = cmd.parameter[0]
+		cmd.command = cmd.parameter[1]
+		cmd.parameter = cmd.parameter.drop(2)
+	}
+
+	def encapsulatedCommand = cmd.encapsulatedCommand(commandClassVersions)
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand, cmd.sourceEndPoint)
+	} else {
+		logDebug "Unable to get encapsulated command: $cmd"
+	}
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCmd = cmd.encapsulatedCommand(commandClassVersions)
+	if (encapsulatedCmd) {
+		zwaveEvent(encapsulatedCmd)
+	} else {
+		log.warn "Unable to extract encapsulated cmd from $cmd"
+	}
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.configurationv1.ConfigurationReport cmd) {
+	runIn(4, refreshSyncStatus)
+	String name = configParams.find { name, param -> param.num == cmd.parameterNumber }?.key
+	if (name) {
+		int val = cmd.scaledConfigurationValue
+		state[name] = val
+		logDebug "${configParams[name]?.title}(#${configParams[name]?.num}) = ${val}"
+	} else {
+		logDebug "Parameter #${cmd.parameterNumber} = ${cmd.scaledConfigurationValue}"
+	}
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.versionv1.VersionReport cmd) {
+	logDebug "${cmd}"
+	sendEvent(name: "firmwareVersion", value: (cmd.applicationVersion + (cmd.applicationSubVersion / 100)))
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd, endpoint=0) {
+	logDebug "${cmd} (${endpoint})"
+	sendSwitchEvents(cmd.value, endpoint)
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.switchbinaryv1.SwitchBinaryReport cmd, endpoint=0) {
+	logDebug "${cmd} (${endpoint})"
+	sendSwitchEvents(cmd.value, endpoints.relay)
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelReport cmd, endpoint=0) {
+	logDebug "${cmd} (${endpoint})"
+	sendSwitchEvents(cmd.value, endpoints.dimmer)
+}
```

Comment:
Is there a possibility that a `SwitchBinaryReport`/`SwitchMultilevelReport` would come from unexpected endpoint?

Reply:
No it's not.  The device always sends the same type of report and if you send a get command to the opposite endpoint it ignores it...

Reply:
In that case do you mind passing an actual `endpoint` which is included in `MultiChannelCmdEncap` instead of those global values?

Reply:
I've made the change.